pkg/cortex/serve/cortex_internal/lib/storage/s3.py

```diff
@@ -97,16 +97,20 @@ def _get_matching_s3_objects_generator(self, prefix="", suffix=""):
 
             for obj in contents:
                 key = obj["Key"]
-                if key.startswith(prefix) and key.endswith(suffix):
+                if (
+                    key.startswith(prefix)
+                    and key.endswith(suffix)
+                    and (include_dir_objects or (not include_dir_objects and not key.endswith("/")))
```

Comment:
nit: I think `not include_dir_objects and` can be removed, e.g:

`and (include_dir_objects or not key.endswith("/"))`

Reply:
Yep, I agree.